File: 프로그래머스/2/42586. 기능개발/기능개발.go
func solution(progresses []int, speeds []int) []int {
	result := []int{}

	for len(progresses) > 0 {
		cnt := 0
		for idx := range progresses {
			progresses[idx] += speeds[idx]
		}
		for len(progresses) != 0 && progresses[0] >= 100 {
			progresses = progresses[1:]
			speeds = speeds[1:]
			cnt++
		}
		if cnt != 0 {
			result = append(result, cnt)
		}
	}

	return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    result = []
    progresses = deque(progresses)
    speeds = deque(speeds)
    while progresses:
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        c = 0
        while progresses and progresses[0] >= 100:
            progresses.popleft()
            speeds.popleft()
            c += 1
        if c: result.append(c)
    return result
            